bin/generate_unfiltered_sce.R

```diff
@@ -161,11 +161,19 @@ sample_metadata_df <- readr::read_tsv(opt$sample_metadata_file) |>
   dplyr::rename("sample_id" = "scpca_sample_id") |>
   # add library ID as column in sample metadata
   # we need this so we are able to merge sample metadata with colData later
-  dplyr::mutate(library_id = opt$library_id) |>
-  # remove upload date as we don't provide this on the portal
-  dplyr::select(-upload_date) |>
-  # rename to donor id for czi compliance
-  dplyr::rename("donor_id" = "participant_id")
+  dplyr::mutate(library_id = opt$library_id) 
+
+if("upload_date" %in% colnames(sample_metadata_df)){
+  sample_metadata_df -> sample_metadata_df |>
+    # remove upload date as we don't provide this on the portal
+    dplyr::select(-upload_date)
+}
+
+if("participant_id" %in% colnames(sample_metadata_df)){
+  sample_metadata_df -> sample_metadata_df |>
+    # rename to donor id for czi compliance
+    dplyr::rename("donor_id" = "participant_id")
+}
```

Comment:
Your assignment operators appear to be pointing the wrong way here? I'm not sure I understand how this worked?

I downloaded `scpca_out` and in unfiltered SCE I see this - 
```
> metadata(sce)$sample_metadata
  sample_id optional_column_1 optional_column_2 library_id
1  sample01        diagnosis1              age1  library01
```

But still, you probably mean:

```suggestion
if("upload_date" %in% colnames(sample_metadata_df)){
  sample_metadata_df <- sample_metadata_df |>
    # remove upload date as we don't provide this on the portal
    dplyr::select(-upload_date)
}

if("participant_id" %in% colnames(sample_metadata_df)){
  sample_metadata_df <- sample_metadata_df |>
    # rename to donor id for czi compliance
    dplyr::rename("donor_id" = "participant_id")
}
```

Reply:
Yikes this is embarrassing. Thank you for catching it! Those columns aren't present in the example sample metadata because they aren't required, so it didn't actually run that code when I ran the example metadata.